akka-cluster-sharding-typed/src/test/java/jdocs/delivery/ShardingDocExample.java

```diff
@@ -0,0 +1,385 @@
+      public final String item;
+
+      public AddTask(String item) {
+        this.item = item;
+      }
+    }
+
+    public static class CompleteTask implements Command {
+      public final String item;
+
+      public CompleteTask(String item) {
+        this.item = item;
+      }
+    }
+
+    private static class InitialState implements Command {
+      final State state;
+
+      private InitialState(State state) {
+        this.state = state;
+      }
+    }
+
+    private static class SaveSuccess implements Command {
+      final ActorRef<ConsumerController.Confirmed> confirmTo;
+
+      private SaveSuccess(ActorRef<ConsumerController.Confirmed> confirmTo) {
+        this.confirmTo = confirmTo;
+      }
+    }
+
+    private static class DBError implements Command {
+      final Exception cause;
+
+      private DBError(Throwable cause) {
+        if (cause instanceof Exception) this.cause = (Exception) cause;
+        else this.cause = new RuntimeException(cause.getMessage(), cause);
+      }
+    }
+
+    private static class CommandDelivery implements Command {
+      final Command command;
+      final ActorRef<ConsumerController.Confirmed> confirmTo;
+
+      private CommandDelivery(Command command, ActorRef<ConsumerController.Confirmed> confirmTo) {
+        this.command = command;
+        this.confirmTo = confirmTo;
+      }
+    }
+
+    public static class State {
+      public final List<String> tasks;
+
+      public State(List<String> tasks) {
+        this.tasks = Collections.unmodifiableList(tasks);
+      }
+
+      public State add(String task) {
+        ArrayList<String> copy = new ArrayList<>(tasks);
+        copy.add(task);
+        return new State(copy);
+      }
+
+      public State remove(String task) {
+        ArrayList<String> copy = new ArrayList<>(tasks);
+        copy.remove(task);
+        return new State(copy);
+      }
+    }
+
+    public static Behavior<Command> create(
+        String id, DB db, ActorRef<ConsumerController.Start<Command>> consumerController) {
+      return Behaviors.setup(
+          context -> {
+            context.pipeToSelf(
+                db.load(id),
+                (state, exc) -> {
+                  if (exc == null) return new InitialState(state);
+                  else return new DBError(exc);
+                });
+
+            return Behaviors.receive(Command.class)
+                .onMessage(
+                    InitialState.class,
+                    initial -> {
+                      ActorRef<ConsumerController.Delivery<Command>> deliveryAdapter =
+                          context.messageAdapter(
+                              ConsumerController.deliveryClass(),
+                              d -> new CommandDelivery(d.message(), d.confirmTo()));
+                      consumerController.tell(new ConsumerController.Start<>(deliveryAdapter));
+
+                      return new Active(context, id, db, initial.state);
+                    })
```

Comment:
A little bit tricky to follow with the nested lambdas and FP-and-then-OO style state, I can see that I would do the same but maybe worth doing an `Initial` `AbstractBehavior` class to make it easier for the Java crowd?

Reply:
ok, changed several of them to AbstractBehavior in https://github.com/akka/akka/pull/28155/commits/dba92f361636805efe4dd8f9b40982685dface95